### DISCUSSION TITLE
Various call sites are only available on certain platforms

### DISCUSSION THREAD

**Discussion by TheBrambleShark** [unoplatform/uno#11624] [Q&A]
I was directed here for this question.

I will occasionally get warnings from Visual Studio stating that certain calls or elements are only available on certain platforms, but are accessible by platforms other than what's allowed.

Some of these I'm completely certain I can ignore. For instance, `Main.iOS.cs` has a warning on the line `UIApplication.Main(args, null, typeof(AppHead));` that warns me the call site is reachable on Android, but Android will never see this class so it's not a concern of mine.

Others are less obvious. For instance, `Frame` elements report only being available on Windows even though it's reachable on Android. This one is more concerning if it's not available since it's my primary navigation method. This code comes from my MainPage:
```xml
<NavigationView x:Name="MainNavView"
                PaneDisplayMode="LeftCompact"
                IsBackEnabled="False"
                SelectionChanged="MainNavView_SelectionChanged"
                Loaded="MainNavView_Loaded"
                Grid.Row="1">
    <NavigationView.MenuItems>
        <NavigationViewItem Icon="Home" Content="Home" Tag="home" />
        <NavigationViewItem Icon="OtherUser" Content="Private Chats" Tag="conversations" />
        <NavigationViewItem Icon="SyncFolder" Content="Communities" Tag="communities"/>
    </NavigationView.MenuItems>

    <NavigationView.FooterMenuItems>
        <NavigationViewItem Icon="Admin" Content="Administration" Tag="admin" />
    </NavigationView.FooterMenuItems>

    <Frame x:Name="MainNavContentFrame" IsTabStop="True" NavigationFailed="MainNavContentFrame_NavigationFailed" />
</NavigationView>
```

Since it's not clear, when should I trust Uno to magic these errors away and when should I actually be concerned?

**Comment by jeromelaban** [comment]
Thanks for the report.

Can you provide a screenshot of what you're seeing? Depending on the active frameworks, some of those warnings are emitted by the compiler or IDE without good causes.

**Reply by TheBrambleShark** [reply]
Attached is a screenshot Main.iOS.cs with the corresponding error. I am also using the latest beta version of StyleCop.Analyzers and my ruleset is here: (the project where I'm having issue is open source, owned by me)
https://github.com/Selkhound/Selkhound.Client/blob/main/src/stylecop.ruleset

![image](https://user-images.githubusercontent.com/7003081/223897286-67bc997b-249c-488e-a6e1-d4824bb2bdab.png)

Edit: Exact text of the error-
```
CA1416 - This call site is reachable on: 'Android' 21.0 and later. 'UIApplication.Main(string[]?, Type?, Type?)' is only supported on: 'ios', 'maccatalyst', 'tvos'.
```

**Reply by jeromelaban** [reply]
This is interesting. In practice, the compile items inclusions are not supposed to be providing this particular file to android (nor should it compile). 

If you take a look at the top left project selector for this particular file, what is in the list ? Here's what you should get:
![image](https://user-images.githubusercontent.com/5839577/223898625-63241515-91fc-4bf0-9c4e-0eed617cf87b.png)

If you have more than `net6.0-ios` or `net7.0-ios` there's a project template error. If not, this would indicate an analyzers overreach of some kind.

**Reply by TheBrambleShark** [reply]
Mine only shows `net6.0-ios`, so it should be fine in that regard.

If you import StyleCop.Analyzers version 1.2.0-beta.435 into your project and use my linked stylecop.ruleset, does the CA1416 error appear for you as well?

**Reply by jeromelaban** [reply]
I am not seeing the error you are seeing. This still seems like an analyzer error, where the file is mistakenly included in a TargetFramework that is not using this specific file.
